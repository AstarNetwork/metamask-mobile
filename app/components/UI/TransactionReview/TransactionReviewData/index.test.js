import React from 'react';
import TransactionReviewData from './';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';

const mockStore = configureMockStore();
const initialState = {
	engine: {
		backgroundState: {
			TokenRatesController: {
				contractExchangeRates: {}
			},
			CurrencyRateController: {
				currentCurrency: 'usd',
				conversionRate: 0.1
			}
		}
	},
	transaction: {
		transaction: {
			data: ''
		},
		value: '',
		from: '0x1',
		gas: '',
		gasPrice: '',
		to: '0x2',
		selectedAsset: undefined,
		assetType: undefined
	}
};
const store = mockStore(initialState);

describe('TransactionReviewData', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<Provider store={store}>
				<TransactionReviewData transactionData={{ amount: 0, gas: 0, gasPrice: 1, from: '0x0' }} />
			</Provider>
		);
		expect(wrapper.dive()).toMatchSnapshot();
	});
});
