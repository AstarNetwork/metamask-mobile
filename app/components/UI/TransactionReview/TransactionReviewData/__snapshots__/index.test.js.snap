// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionReviewData should render correctly 1`] = `
<Connect(TransactionReviewData)
  transactionData={
    Object {
      "amount": 0,
      "from": "0x0",
      "gas": 0,
      "gasPrice": 1,
    }
  }
/>
`;
